api/v1beta1/openstackprovisionserver.go

```diff
@@ -40,7 +40,7 @@ func AssignProvisionServerPort(
 	instance *OpenStackProvisionServer,
 	portStart int32,
 ) error {
-	if instance.Spec.Port != 0 {
```

Comment:
Not sure I understand. AssignProvisionServerPort() is called from baremetalset controller and not from provisionserver controller. This is done when provisonserver is created/patched.

[1] https://github.com/openstack-k8s-operators/openstack-baremetal-operator/blob/master/controllers/openstackbaremetalset_controller.go#L474-L481